icevision/models/ross/efficientdet/dataloaders.py

```diff
@@ -163,16 +159,41 @@ def build_infer_batch(dataset, batch_tfms=None):
     outs = model(*batch)
     ```
     """
-    samples = common_build_batch(dataset, batch_tfms=batch_tfms)
+    records = common_build_batch(records, batch_tfms=batch_tfms)
+    batch_images, batch_sizes, batch_scales = zip(
+        *(process_infer_record(record) for record in records)
+    )
+
+    # convert to tensors
+    batch_images = torch.stack(batch_images)
+    batch_sizes = Tensor(batch_sizes)
+    batch_scales = Tensor(batch_scales)
+
+    # convert to EffDet interface
+    targets = dict(
+        img_size=batch_sizes,
+        img_scale=batch_scales,
+    )
+
+    return (batch_images, targets), records
+
+
+def process_train_record(record) -> tuple:
+    """Extracts information from record and prepares a format required by the EffDet training"""
+    image = im2tensor(record["img"])
+    classes = (
+        record["labels"] if record["labels"] else [0]
```

Comment:
I think we should use `len(record['labels'] > 0)` to be more strict and avoid possible bugs. 

For example, what happens if for some reason `record['labels'] = None`? It's not something we are prepared for and it would mean something wrong is happening in the pipeline, it's better for that to error out here.

Reply:
you're right this looks cleaner but at the same time is more error prone